presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriteUtils.java

```diff
@@ -458,7 +459,8 @@ public static boolean pathExists(HdfsContext context, HdfsEnvironment hdfsEnviro
     public static boolean isS3FileSystem(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
     {
         try {
-            return getRawFileSystem(hdfsEnvironment.getFileSystem(context, path)) instanceof PrestoS3FileSystem;
+            FileSystem fileSystem = getRawFileSystem(hdfsEnvironment.getFileSystem(context, path));
+            return fileSystem instanceof PrestoS3FileSystem || fileSystem.getClass().getName().equals(HiveS3Module.EMR_FS_CLASS_NAME);
```

Comment:
Nit: static import the `EMR_FS_CLASS_NAME` constant since the `HiveS3Module.` doesn't provide any value (and thus having it makes the code harder to read)

Reply:
Thanks for the feedback, I've updated the code to your suggestion.